src/test/java/spoon/test/method/MethodTest.java

```diff
@@ -218,4 +220,16 @@ public void test_addFormalCtTypeParameterAt_throwsOutOfBoundsException_whenPosit
 		assertThrows(IndexOutOfBoundsException.class,
 				() -> method.addFormalCtTypeParameterAt(1, typeParam));
 	}
+
+	@Test
+	public void testGetAllMethodsInnerClassExtended() {
+		// contract: implicit static nested interfaces are correct handled in getAllExecutables and dont throw an error.
+		Launcher launcher = new Launcher();
+		launcher.addInputResource("src/test/resources/extendsStaticInnerType");
+		CtModel model = launcher.buildModel();
+		CtType<?> type = model.getAllTypes().stream().filter(v -> v.getSimpleName().contains("BarBaz")).findAny().get();
+		assertDoesNotThrow(() -> type.getAllExecutables());
+		// on jdk8 there are 14 types on newer 13 because a method in Object.java was removed
+		assertTrue(13 == type.getAllExecutables().size() || 14 == type.getAllExecutables().size());
```

Comment:
I have two comments on these assertions.

First of all, `assertDoesNotThrow` is redundant as, if `type.getAllExecutables()` throws an exception, then the `assertTrue` will surely fail. So, if the test is meant to verify only that `getAllExecutables()` does not throw, then keep only the `assertDoesNotThrow` assertion. If the test is meant to verify the amount of returned executables (which is a stronger oracle than that it does not throw), then keep only the `assertTrue`.

Second, I strongly advice against using `assertTrue` for anything but actual boolean values, because when the test fails you only get "false", and not what the values actually were. If the test should verify the amount of returned executables, then I would refactor it like so:

```suggestion
		int expectedNumExecutablesInJDK8 = 13;
		int expectedNumExecutablesPostJDK8 = 14;

		int numExecutables = type.getAllExecutables().size();

		assertThat(numExecutables, anyOf(
				equalTo(expectedNumExecutablesInJDK8),
				equalTo(expectedNumExecutablesPostJDK8))
		);
```

Reply:
Good suggestion and way better than my abuse of `assertTrue`